src/link.js

```diff
@@ -34,3 +34,150 @@ export default class Link extends Plugin {
 		return 'Link';
 	}
 }
+
+/**
+ * The configuration of the {@link module:link/link~Link} feature.
+ *
+ * Read more in {@link module:link/link~LinkConfig}.
+ *
+ * @member {module:link/link~LinkConfig} module:core/editor/editorconfig~EditorConfig#link
+ */
+
+/**
+ * The configuration of the {@link module:link/link~Link link feature}.
+ *
+ *		ClassicEditor
+ *			.create( editorElement, {
+ * 				link:  ... // Link feature configuration.
+ *			} )
+ *			.then( ... )
+ *			.catch( ... );
+ *
+ * See {@link module:core/editor/editorconfig~EditorConfig all editor options}.
+ * @interface LinkConfig
+ */
+
+/**
+ * When set `true`, the `target="blank"` and `rel="noopener noreferrer"` attributes are automatically added to all external links
+ * in the editor. By external are meant all links in the editor content starting with `http`, `https`, or `//`.
+ *
+ * Internally, this option activates a predefined {@link module:link/link~LinkConfig#decorators automatic link decorator},
+ * which extends all external links with the `target` and `rel` attributes without additional configuration.
+ *
+ * **Note**: To control the `target` and `rel` attributes of specific links in the edited content, a dedicated
+ * {@link module:link/link~LinkDecoratorManualOption manual} decorator must be defined in the
+ * {@link module:link/link~LinkConfig#decorators `config.link.decodators`} array. In such scenario,
+ * the `config.link.targetDecorator` option should remain `undefined` or `false` to not interfere with the manual decorator.
+ *
+ * **Note**: It is possible to add other {@link module:link/link~LinkDecoratorAutomaticOption automatic}
+ * or {@link module:link/link~LinkDecoratorManualOption manual} link decorators when this option is active.
+ *
+ * More information about decorators can be found in the {@link module:link/link~LinkConfig#decorators decorators configuration}
+ * reference.
+ *
+ * @default false
+ * @member {Boolean} module:link/link~LinkConfig#targetDecorator
+ */
+
+/**
+ * Decorators provide an easy way to configure and manage additional link attributes in the editor content. There are
+ * two types of link decorators:
+ *
+ * * {@link module:link/link~LinkDecoratorAutomaticOption automatic} – they match links against pre–defined rules and
+ * manage their attributes based on the results,
+ * * {@link module:link/link~LinkDecoratorManualOption manual} – they allow users to control link attributes individually
+ *  using the editor UI.
+ *
+ * Link decorators are defined as an array of objects:
+ *
+ *		const linkConfig = {
+ *			decorators: [
+ *				{
+ *					mode: 'automatic',
+ *					callback: url => url.startsWith( 'http://' ),
+ *					attributes: {
+ *						target: '_blank',
```

Comment:
How to configure a value-less HTML attribute? 

For instance, if the [`download`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Attributes) has a value, it is reflected in the name of the downloaded file. So we cannot use `download: true` here safely because it will download the file named "true".

Reply:
This case as we discuss is rather engine issue, and should be extracted as separate. If `engine.view.elemendefinition` start support attributes without value, decorators will be able to use it as well.

Reply:
Extracted to:
https://github.com/ckeditor/ckeditor5-engine/issues/1750